main_test.go

```diff
@@ -95,3 +95,60 @@ func TestCopyFromCommandLineArgument(t *testing.T) {
 		t.Errorf("Expected %s but got %s", content, copied)
 	}
 }
+
+func TestParseContentDirectText(t *testing.T) {
+	directText := "Direct text input"
+	contentStr, err := parseContent(&directText, []string{})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != directText {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", directText, contentStr)
+	}
+}
+
+func TestParseContentStdin(t *testing.T) {
+	stdinInput := "Stdin input\n"
+	r, w, _ := os.Pipe()
+	originalStdin := os.Stdin
+	defer func() {
+		os.Stdin = originalStdin
+		r.Close()
+		w.Close()
+	}()
+	os.Stdin = r
+
+	// Write the input to the pipe
+	w.WriteString(stdinInput)
+	w.Close()
+
+	contentStr, err := parseContent(nil, []string{})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != stdinInput {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", stdinInput, contentStr)
+	}
+}
+
+func TestParseContentFromFile(t *testing.T) {
+	contentFromFile := "Content from file"
+	fakeFile := createTempFile(contentFromFile)
+	defer os.Remove(fakeFile.Name())
+
+	contentStr, err := parseContent(nil, []string{fakeFile.Name()})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != contentFromFile+"\n" {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", contentFromFile, contentStr)
+	}
+}
+
+// Helper function to create temporary file for testing
+func createTempFile(content string) *os.File {
```

Comment:
Here is a safer and cleaner way to do


```suggestion
func createTempFile(t *testing.T, content string) *os.File {

         file, err := os.CreateTemp("", "testfile")
         if err != nil {
              t.Fail("creating temporary test file")
         }

         // register the cleaner without having a need to handle the file removal with a defer that can be forgotten
         t.Cleanup (func() {
                        defer os.Remove(file.Name())
         })
         file.WriteString(content) 
         file.Close() 
         return file
}

```

If you have a look at os.CreateTemp code

https://cs.opensource.google/go/go/+/go1.22.4:src/os/tempfile.go;l=33

You will see the code is using tempdir when using "" as empty folder.

So you are creating a folder, but nothing delete it when the test are over.

Then, now you have a better understanding, there is the better way to write it

```suggestion
func createTempFile(t *testing.T, content string) *os.File {

         file, err := os.CreateTemp(t.TempDir(), "testfile")
         if err != nil {
              t.Fail("creating temporary test file", err)
         }

         file.WriteString(content) 
         file.Close() 
         return file
}
```

Please have look at https://pkg.go.dev/testing#T.TempDir

and its implementation https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/testing/testing.go;l=1229

No need for cleaning up the file, then because the whole temp folder is nuked by the cleanup added via `t.TempDir`

Reply:
> Here is a safer and cleaner way to do

I opened an issue in an [existing linter](https://github.com/peczenyj/ttempdir/issues/19) to report the use case I discovered [here](https://github.com/supitsdu/clipper/pull/17#discussion_r1651958704)